doc/source/run_engine_api.rst

```diff
@@ -0,0 +1,56 @@
+.. currentmodule:: bluesky.run_engine
+
+RunEngine API Documentation
+===========================
+
+.. autoclass:: RunEngine
+
+    .. automethod:: __call__
+
+    The RunEngine maintains a callback registry of functions that receive any
+    :doc:`documents` generated by plan execution. These methods add and remove
+    functions from that registry.
+
+    .. automethod:: subscribe
```

Comment:
OK, I'm up for that, but this feels like the beginning of a three-hour fight with sphinx that I don't have time for before Chicago. Punting to https://github.com/NSLS-II/bluesky/issues/701